server/src/main/kotlin/delta/codecharacter/server/user/public_user/PublicUserService.kt

```diff
@@ -71,11 +69,50 @@ class PublicUserService(@Autowired private val publicUserRepository: PublicUserR
                     wins = it.wins,
                     losses = it.losses,
                     ties = it.ties
-                )
+                ),
             )
         }
     }
 
+    fun updateLeaderboard(leaderboardList: List<LeaderboardEntryDto>) {
+        val leaderboardSize = leaderboardList.size
+        for (i in 0 until leaderboardSize) {
+            val user = publicUserRepository.findByUsername(leaderboardList[i].user.username).get()
```

Comment:
- Querying the publicuser entity and changing to DTOs and then querying same entity by user name seems to many redudant queries.
- For updating leaderboard, query the entire users and then update accordingly, it will be easier to do